### PR TITLE
Fixed Scala Doc Warnings

### PR DESCRIPTION
**Pull request by bnasim**
Some of the ScalaDoc Warnings has been fixed.

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @bnasim,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.typesafe.com/contribute/cla](http://www.typesafe.com/contribute/cla)

File: framework/src/play/src/main/scala/play/core/parsers/FormUrlEncodedParser.scala [hunk 1]
```diff
@@ -37,9 +37,9 @@ object FormUrlEncodedParser {
 
   /**
    * Parse the content type "application/x-www-form-urlencoded", mapping to a Java compatible format.
-   * @param data
-   * @param encoding
-   * @return
+   * @param data The body content of the request, or whatever needs to be so parsed
+   * @param encoding The character encoding of data
+   * @return A ListMap of keys to the sequence of values for that key
```
Comment (jroper) [inline framework/src/play/src/main/scala/play/core/parsers/FormUrlEncodedParser.scala]: It's not actually a ListMap, it should just say Map.

**Comment by marcospereira** [comment]
@bnasim besides the comment made by @jroper, could you please also squash your commits into a single one?

**Comment by bnasim** [comment]
I squashed my commits.

File: framework/src/play/src/main/scala/play/core/parsers/FormUrlEncodedParser.scala (hunk 1 above)
Comment (gmethvin) [inline framework/src/play/src/main/scala/play/core/parsers/FormUrlEncodedParser.scala]: This should probably also mention somewhere the values are arrays as opposed to lists, which appears to be the main difference between this and the `parseAsJava` method.

**Comment by gmethvin** [comment]
@bnasim Please fix the documentation issue from the comment above and we'll merge.

**Comment by marcospereira** [comment]
@bnasim Thank you very much for your time. We are almost there. :-)

This pull request now have some conflicts. Could you please rebase with master and squash the commits? After that, I think this PR will be ready to be merged.

**Pull request closed without merging**